### PR TITLE
DPAD-1089  :: Youtube Embed Player

### PR DESCRIPTION
**Pull request by p-forfa**
- Added in changes to allow Youtube embed player to load on the Desktop and Mobile version of the video player
- Converted all Twitch takeover utilities and components into Youtube embed components
- Added in query param to force the Youtube embed to show up, if a proper youtube video id is passed in
- Added in direct network calls to the article-video service to quickly check whether a youtube video should be turned on/off quickly, instead of relying on the data that comes back from the wiki Video Details (which have about a 3-24 hour cache time)
- Skipped logic for checking whether the Youtube embed should occur, if the wiki is a tier 3 wiki. This ensures that network calls ONLY occur for wikis that are targeted, resulting in less overall network calls. In the end, a youtube embed will only occur for targeted wikis, never for tier 3 wikis.

### CODE REVIEW

File: test-jw/src/videoConfigs.js [hunk 1]
```diff
@@ -6608,5 +6608,6 @@ export const ARTICLE_VIDEO_DETAILS = {
                 "fcc4038c": 0.7692186749171301
             }
         }
-    }
+    },
+    "tier3Mapping": false,
 }
```
Comment (gundams-and-dragons) [inline test-jw/src/videoConfigs.js:6613]: Missing newline at the end of the file.

File: src/loaders/utils/GetYoutubeTakeoverDetails.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	if (articleVideoBaseUrl.length === 0) {
+		return articleVideoBaseUrl;
+	}
+	if (!wikiId) {
+		return '';
+	}
+	return `${articleVideoBaseUrl}youtube/v1/youtube-takeover-mappings/${wikiId}`;
+}
+
+interface MWConfig {
```
Comment (gundams-and-dragons) [inline src/loaders/utils/GetYoutubeTakeoverDetails.ts:17]: Should these be moved into `src/youtube/types.d.ts` ?

File: src/youtube/players/YoutubeDesktopArticleVideoPlayer.tsx [hunk 3]
```diff
@@ -59,34 +58,31 @@ const DesktopArticleVideoWrapper = styled.div<DesktopArticleVideoWrapperProps>`
...
-const DesktopArticleVideoPlayer: React.FC = () => {
+const YoutubeDesktopArticleVideoPlayer: React.FC<YoutubeArticleVideoPlayerProps> = ({ youtubeTakeoverDetails }) => {
 	const placeholderRef = useRef<HTMLDivElement>(null);
 	const onScreen = useOnScreen(placeholderRef, '0px', 0.1);
-	// const [dismissed, setDismissed] = useState(false);
-	// const isScrollPlayer = !(dismissed || onScreen);
-	const isScrollPlayer = !onScreen;
+	const [dismissed, setDismissed] = useState(false);
+	const isScrollPlayer = !(dismissed || onScreen);
+	// const isScrollPlayer = !onScreen;
```
Comment (gundams-and-dragons) [inline src/youtube/players/YoutubeDesktopArticleVideoPlayer.tsx:71]: Still need this or good to remove it?

**Review by gundams-and-dragons — CHANGES_REQUESTED** [review]

File: test-jw/src/videoConfigs.js (hunk 1 above)
Comment (p-forfa) [reply test-jw/src/videoConfigs.js]: Fixed in fcc33538c5143985c81ccbc7bd821c5193df6c63

File: src/loaders/utils/GetYoutubeTakeoverDetails.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	}
+
+	return youtubeTakeoverDetails;
+}
+
+export const eligibleForYoutubeTakeover = (youtubeTakeoverDetails: YoutubeTakeOverDetails) => {
+	const youtubeTakeoverFlag = youtubeTakeoverDetails.isYoutubeTakeover;
+	const isVideoIdValidLength =
+		youtubeTakeoverDetails?.youtubeVideoId && youtubeTakeoverDetails?.youtubeVideoId?.length !== 0;
+	console.log(
```
[the remark this answers is not in the document]
Comment (p-forfa) [reply src/loaders/utils/GetYoutubeTakeoverDetails.ts]: Fixed in 7f0eacb7a5277a767971c6315f12b717de6954cf

File: src/loaders/utils/GetYoutubeTakeoverDetails.ts (hunk 2 above)
Comment (p-forfa) [reply src/loaders/utils/GetYoutubeTakeoverDetails.ts]: Fixed in 954b9d45657bcecabad3eb63bf4152415f424650

File: src/youtube/players/YoutubeDesktopArticleVideoPlayer.tsx (hunk 3 above)
Comment (p-forfa) [reply src/youtube/players/YoutubeDesktopArticleVideoPlayer.tsx]: Fixed in 46fc2fabe41513a843613e8bf199f6abf9bb8607

**Review by gundams-and-dragons — APPROVED** [review]

**Pull request merged**